lizmap/modules/lizmap/lib/Project/ProjectMainData.php

```diff
@@ -0,0 +1,562 @@
+            && property_exists($cfgContent->options, 'projection')
+            && property_exists($cfgContent->options->projection, 'ref')) {
+            return $cfgContent->options->projection->ref;
+        }
+
+        return '';
+    }
+
+    /**
+     * Get option bbox in a Lizmap config content.
+     *
+     * @param object $cfgContent - the Lizmap config content
+     *
+     * @return string
+     */
+    protected function getBboxOption($cfgContent)
+    {
+        if (property_exists($cfgContent, 'options')
+            && property_exists($cfgContent->options, 'bbox')) {
+            return implode(', ', $cfgContent->options->bbox);
+        }
+
+        return '';
+    }
+
+    /**
+     * Check if the project needs an update which lead to an error.
+     *
+     * @param object $cfgContent               - the Lizmap config content
+     * @param int    $requiredTargetLwcVersion - the configured required target Lizmap Web Client version
+     *
+     * @return bool true if the project needs to be updated in the QGIS desktop plugin
+     */
+    protected function getNeedsUpdateError($cfgContent, $requiredTargetLwcVersion)
+    {
+        $lizmapWebClientTargetVersion = 30200;
+        if (property_exists($cfgContent, 'metadata')
+            && property_exists($cfgContent->metadata, 'lizmap_web_client_target_version')) {
+            $lizmapWebClientTargetVersion = $cfgContent->metadata->lizmap_web_client_target_version;
+        }
+
+        return $lizmapWebClientTargetVersion < $requiredTargetLwcVersion;
+    }
+
+    /**
+     * Check acl rights on the project.
+     *
+     * @param object                  $cfgContent - the Lizmap config content
+     * @param App\AppContextInterface $appContext - the application context
+     *
+     * @return bool true if the current user as rights on the project
+     */
+    protected function checkAcl($cfgContent, $appContext)
+    {
+        // Check acl option is configured in project config
+        $aclGroups = null;
+        if (property_exists($cfgContent, 'options')
+            && property_exists($cfgContent->options, 'acl')) {
+            $aclGroups = $cfgContent->options->acl;
+        }
+        if ($aclGroups === null || !is_array($aclGroups) || empty($aclGroups)) {
+            return true;
+        }
+
+        // Check user is authenticated
+        if (!$appContext->userIsConnected()) {
+            return false;
+        }
+
+        // Check if configured groups white list and authenticated user groups list intersects
+        $userGroups = $appContext->aclUserGroupsId();
+        if (array_intersect($aclGroups, $userGroups)) {
+            return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * Get option hide project in a Lizmap config content.
+     *
+     * @param object $cfgContent - the Lizmap config content
+     *
+     * @return bool
+     */
+    protected function getHiddenOption($cfgContent)
```

Comment:
We can use explicit return type (at least for simple return type ?) 
 ```suggestion
     protected function getHiddenOption($cfgContent): bool
```